messaging/src/main/java/org/axonframework/eventhandling/deadletter/EvaluationTask.java

```diff
@@ -0,0 +1,130 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventhandling.deadletter;
+
+import org.axonframework.common.transaction.TransactionManager;
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.eventhandling.EventMessageHandler;
+import org.axonframework.eventhandling.ListenerInvocationErrorHandler;
+import org.axonframework.messaging.deadletter.DeadLetter;
+import org.axonframework.messaging.deadletter.DeadLetterEvaluationException;
+import org.axonframework.messaging.deadletter.DeadLetterQueue;
+import org.axonframework.messaging.unitofwork.DefaultUnitOfWork;
+import org.axonframework.messaging.unitofwork.UnitOfWork;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.invoke.MethodHandles;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * A {@link Runnable} implementation used to evaluate a {@link DeadLetter} taken from the {@link DeadLetterQueue}. This
+ * task is added through {@link DeadLetterQueue#onAvailable(String, Runnable)}, so we can typically assume there are
+ * entries ready for evaluation.
+ *
+ * @author Steven van Beelen
+ * @since 4.6.0
+ */
+class EvaluationTask implements Runnable {
+
+    private final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    private final List<EventMessageHandler> eventHandlingComponents;
+    private final DeadLetterQueue<EventMessage<?>> queue;
+    private final String processingGroup;
+    private final TransactionManager transactionManager;
+    private final ListenerInvocationErrorHandler listenerInvocationErrorHandler;
+
+    EvaluationTask(List<EventMessageHandler> eventHandlingComponents,
+                   DeadLetterQueue<EventMessage<?>> queue,
+                   String processingGroup,
+                   TransactionManager transactionManager,
+                   ListenerInvocationErrorHandler listenerInvocationErrorHandler) {
+        this.eventHandlingComponents = eventHandlingComponents;
+        this.queue = queue;
+        this.processingGroup = processingGroup;
+        this.transactionManager = transactionManager;
+        this.listenerInvocationErrorHandler = listenerInvocationErrorHandler;
+    }
+
+    @Override
+    public void run() {
+        AtomicBoolean evaluationFailed = new AtomicBoolean(false);
+        Optional<DeadLetter<EventMessage<?>>> optionalLetter;
+        while (!evaluationFailed.get() &&
```

Comment:
Should the `EvaluationTask` stop after a single event failed during evaluation?
Perhaps other sequences can still be taken care of.

Thoughts?

Reply:
I'm afraid we can't do this without implementing an ignore list for sequence numbers. Especially in the database the query will be quite reliable in the ordering, so we would just keep retrying the message

Reply:
Cool, thanks for the suggestion!